docker/docker-compose.yml

```diff
@@ -1,147 +1,145 @@
-version: '2.2'
```

Comment:
Several changes here (worth summarizing at the top to help prevent confusion):
* Update Compose format to 3.8 (latest at this time) from 2.2
* Change env var declarations to be in map form rather than list form (reads easier in VS Code if there isn't a strong preference on style :))
* Remove `container_name` as it has the possibility of being slightly confusing when working with Docker Compose (since the container name is not the same as the service name it's easy to type it in as the target service name)
* Change `restart` policy to `on-failure` which is a bit more specific
* Use software versions as defined in `.env`.